test/e2e/e2e_suite_test.go

```diff
@@ -126,10 +133,53 @@ var _ = BeforeSuite(func() {
 
 var _ = AfterSuite(func() {
 	fmt.Fprintf(GinkgoWriter, "Tearing down kind cluster\n")
+	retrieveAllLogs()
 	kindCluster.Teardown()
 	os.RemoveAll(suiteTmpDir)
 })
 
+func retrieveAllLogs() {
+	capiLogs := retrieveLogs(capiNamespace, capiDeploymentName)
+	capaLogs := retrieveLogs(capaNamespace, capaDeploymentName)
+
+	// If running in prow, output the logs to the artifacts path
+	artifactPath, exists := os.LookupEnv("ARTIFACTS")
+	if exists {
+		ioutil.WriteFile(path.Join(artifactPath, "capi.log"), []byte(capiLogs), 0644)
+		ioutil.WriteFile(path.Join(artifactPath, "capa.log"), []byte(capaLogs), 0644)
+		return
+	}
+
+	fmt.Fprintf(GinkgoWriter, "CAPI Logs:\n%s\n", capiLogs)
+	fmt.Fprintf(GinkgoWriter, "CAPA Logs:\n%s\n", capaLogs)
+}
+
+func retrieveLogs(namespace, deploymentName string) string {
+	deployment := &appsv1.Deployment{}
+	Expect(kindClient.Get(context.TODO(), apimachinerytypes.NamespacedName{Namespace: namespace, Name: deploymentName}, deployment)).To(Succeed())
```

Comment:
You should be able to use `crclient.ObjectKey` which is "aliased" to `NamespacedName`, if you want to avoid the extra import

Reply:
updated, thanks!